apps/src/codebridge/FileBrowser/hooks/useCheckForDuplicateFoldername.ts

```diff
@@ -0,0 +1,27 @@
+import {ProjectFolder} from '@codebridge/types';
+import {useCallback} from 'react';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+
+// Check if the foldername is already in use in the given folder.
+// If it is, alert the user and return true, otherwise return false.
+export const useCheckForDuplicateFoldername = () => {
```

Comment:
It wasn't actually necessary - it was a historical artifact. I just made it into a helper function instead.

Looking through the blames, it looks like the function was originally added inline into `FileBrowser`, but this was triggering extra re-renders along the chain because it was newly created on each render.

Then those functions were retooled into memoized functions to prevent the re-renders, and then on this pass I just yanked those memoized values out into separate files w/o paying attention to them. But it wasn't memoizing any specific data, the memoize was just to prevent the re-renders when it was inline.

Anyway, moved around into functions instead. Good catch!